handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -840,8 +840,25 @@ public final SSLEngineResult wrap(
 
                 // There was no pending data in the network BIO -- encrypt any application data
                 int bytesConsumed = 0;
+                assert bytesProduced == 0;
+
                 // Flush any data that may have been written implicitly by OpenSSL in case a shutdown/alert occurs.
                 bytesProduced = SSL.bioFlushByteBuffer(networkBIO);
+
+                if (bytesProduced > 0) {
```

Comment:
IIUC, `bytesProduced` will be `> 0` only if there is an error from flush

Reply:
no it will be > 0 if there was something pending in the buffer.